File: src/ClaudeUsageWatcher.Core/UsageFormatting.cs
using System.Globalization;

namespace ClaudeUsageWatcher.Core;

public static class UsageFormatting
{
    public static UsageLevel GetLevel(double utilization)
    {
        if (utilization >= 90)
        {
            return UsageLevel.Critical;
        }

        if (utilization >= 70)
        {
            return UsageLevel.Warning;
        }

        return UsageLevel.Normal;
    }

    public static string FormatCountdown(DateTimeOffset? resetsAt, DateTimeOffset now)
    {
        if (resetsAt is null)
        {
            return "";
        }

        TimeSpan delta = resetsAt.Value - now;

        if (delta <= TimeSpan.Zero)
        {
            return "now";
        }

        if (delta >= TimeSpan.FromHours(1))
        {
            int totalHours = (int)Math.Floor(delta.TotalHours);
            int minutes = delta.Minutes;
            return string.Create(CultureInfo.InvariantCulture, $"{totalHours}h {minutes}m");
        }

        if (delta >= TimeSpan.FromMinutes(1))
        {
            int minutes = (int)Math.Floor(delta.TotalMinutes);
            return string.Create(CultureInfo.InvariantCulture, $"{minutes}m");
        }

        return "<1m";
    }
}

File: macos/Sources/UsageCore/UsageFormatting.swift
import Foundation

public enum UsageFormatting {
    public static func level(for utilization: Double) -> UsageLevel {
        if utilization >= 90 {
            return .critical
        }
        if utilization >= 70 {
            return .warning
        }
        return .normal
    }

    /// Mirrors the C# FormatCountdown: invariant "Xh Ym" / "Xm" / "<1m" / "now" / "".
    public static func formatCountdown(_ resetsAt: Date?, now: Date) -> String {
        guard let resetsAt else {
            return ""
        }

        let delta = resetsAt.timeIntervalSince(now)

        if delta <= 0 {
            return "now"
        }

        if delta >= 3600 {
            let totalHours = Int(floor(delta / 3600))
            // Minutes component of the remainder, matching TimeSpan.Minutes (0...59).
            let minutes = Int(floor(delta.truncatingRemainder(dividingBy: 3600) / 60))
            return "\(totalHours)h \(minutes)m"
        }

        if delta >= 60 {
            let minutes = Int(floor(delta / 60))
            return "\(minutes)m"
        }

        return "<1m"
    }
}
